cmd/client-s3.go

```diff
@@ -864,71 +851,75 @@ func (c *s3Client) Stat(isIncomplete bool) (*clientContent, *probe.Error) {
 	if isIncomplete {
 		for objectMultipartInfo := range c.api.ListIncompleteUploads(bucket, object, nonRecursive, nil) {
 			if objectMultipartInfo.Err != nil {
-				return nil, probe.NewError(objectMultipartInfo.Err)
+				return nil, nil, probe.NewError(objectMultipartInfo.Err)
 			}
 
 			if objectMultipartInfo.Key == object {
 				objectMetadata.URL = *c.targetURL
 				objectMetadata.Time = objectMultipartInfo.Initiated
 				objectMetadata.Size = objectMultipartInfo.Size
 				objectMetadata.Type = os.FileMode(0664)
-				return objectMetadata, nil
+				return objectMetadata, nil, nil
 			}
 
 			if strings.HasSuffix(objectMultipartInfo.Key, string(c.targetURL.Separator)) {
 				objectMetadata.URL = *c.targetURL
 				objectMetadata.Type = os.ModeDir
-				return objectMetadata, nil
+				return objectMetadata, nil, nil
 			}
 		}
-		return nil, probe.NewError(ObjectMissing{})
+		return nil, nil, probe.NewError(ObjectMissing{})
 	}
 
 	for objectStat := range c.listObjectWrapper(bucket, object, nonRecursive, nil) {
 		if objectStat.Err != nil {
-			return nil, probe.NewError(objectStat.Err)
+			return nil, nil, probe.NewError(objectStat.Err)
 		}
 
 		if objectStat.Key == object {
 			objectMetadata.URL = *c.targetURL
 			objectMetadata.Time = objectStat.LastModified
 			objectMetadata.Size = objectStat.Size
 			objectMetadata.Type = os.FileMode(0664)
-			return objectMetadata, nil
+			return objectMetadata, map[string][]string{}, nil
 		}
 
 		if strings.HasSuffix(objectStat.Key, string(c.targetURL.Separator)) {
 			objectMetadata.URL = *c.targetURL
 			objectMetadata.Type = os.ModeDir
-			return objectMetadata, nil
+			return objectMetadata, map[string][]string{}, nil
 		}
 	}
 	objectStat, e := c.api.StatObject(bucket, object)
 	if e != nil {
 		errResponse := minio.ToErrorResponse(e)
 		if errResponse.Code == "AccessDenied" {
-			return nil, probe.NewError(PathInsufficientPermission{Path: c.targetURL.String()})
+			return nil, nil, probe.NewError(PathInsufficientPermission{Path: c.targetURL.String()})
 		}
 		if errResponse.Code == "NoSuchBucket" {
-			return nil, probe.NewError(BucketDoesNotExist{
+			return nil, nil, probe.NewError(BucketDoesNotExist{
 				Bucket: bucket,
 			})
 		}
 		if errResponse.Code == "InvalidBucketName" {
-			return nil, probe.NewError(BucketInvalid{
+			return nil, nil, probe.NewError(BucketInvalid{
 				Bucket: bucket,
 			})
 		}
 		if errResponse.Code == "NoSuchKey" || errResponse.Code == "InvalidArgument" {
-			return nil, probe.NewError(ObjectMissing{})
+			return nil, nil, probe.NewError(ObjectMissing{})
 		}
-		return nil, probe.NewError(e)
+		return nil, nil, probe.NewError(e)
 	}
 	objectMetadata.URL = *c.targetURL
 	objectMetadata.Time = objectStat.LastModified
 	objectMetadata.Size = objectStat.Size
 	objectMetadata.Type = os.FileMode(0664)
-	return objectMetadata, nil
+
+	metadata := objectStat.Metadata
```

Comment:
> If metadata can be constructed from *clientContent

Not sure what do you mean, `clientContent` is defined in mc and `metadata` is returned from minio-go. so metadata is not constructed from clientContent.